test/album.js

```diff
@@ -76,4 +77,152 @@ describe('albums', () => {
+        login({
+          email: 'pepito.perez@wolox.com',
+          password: 'Holahola23'
+        }).then(res => {
+          chai
+            .request(server)
+            .post('/albums/1')
+            .set(config.common.session.header_name, res.headers[config.common.session.header_name])
+            .send({
+              id: res.body.id
+            })
+            .then(async result => {
+              const album = await Album.find({
+                where: {
+                  albumId: 1,
+                  userId: 1
+                }
+              });
+              expect(album).to.be.a('object');
+              expect(album.albumId).to.be.equal(1);
+              expect(album.userId).to.be.equal(1);
+              dictum.chai(result, 'buy an album');
+              done();
+            });
+        });
+      });
+    });
+
+    it('should fail because try to buy multiples times same album', done => {
+      const [mockedAlbum] = albums;
+
+      nock(`${url}/1`)
+        .get('')
+        .reply(200, mockedAlbum);
+
+      saveUser(userOne).then(() => {
+        login({
+          email: 'pepito.perez@wolox.com',
+          password: 'Holahola23'
+        }).then(res => {
+          chai
+            .request(server)
+            .post('/albums/1')
+            .set(config.common.session.header_name, res.headers[config.common.session.header_name])
+            .send({
+              id: res.body.id
+            })
+            .then(() => {
+              nock(`${url}/1`)
+                .get('')
+                .reply(200, mockedAlbum);
+
+              chai
+                .request(server)
+                .post('/albums/1')
+                .set(config.common.session.header_name, res.headers[config.common.session.header_name])
+                .send({
+                  id: res.body.id
+                })
+                .catch(err => {
+                  err.should.have.status(400);
+                  err.response.should.be.json;
+                  err.response.body.should.have.property('message');
+                  err.response.body.should.have.property('internal_code');
+                  expect(err.response.body.internal_code).to.equal('album_order_error');
+                  done();
+                });
+            });
+        });
+      });
+    });
+
+    it('should fail because the album does not exist', done => {
+      nock(`${url}/10`)
+        .get('')
+        .reply(404, 'missing album');
+
+      saveUser(userOne).then(() => {
+        login({
+          email: 'pepito.perez@wolox.com',
+          password: 'Holahola23'
+        }).then(res => {
+          chai
+            .request(server)
+            .post('/albums/10')
+            .set(config.common.session.header_name, res.headers[config.common.session.header_name])
+            .send({
+              id: res.body.id
+            })
+            .catch(err => {
+              expect(err.response.request.req.res.body.message).to.be.equal(
```

Comment:
Check response status code, should be 404

Reply:
no, i don't know but always return an 503 code, the difference is the message

Reply:
The message should be that one, check how do you catch error from api